gooey/gui/components/widgets/richtextconsole.py

```diff
@@ -58,6 +61,24 @@ def __init__(self, parent):
             # NB : we use a default parameter to force the evaluation of the binding
             self.actionsMap[escSeq] = lambda bindedColor=wxcolor: self.BeginTextColour(bindedColor)
 
+    def PrepocessAndWriteText(self, content):
+        """Write text into console, while capturing URLs and making 
+        them blue, underlined, and clickable.
+        """
+        l=re.split(self.regex_urls, content)
+        while l:
```

Comment:
I still feel like you're over thinking this a bit with the while loop + manual deletions of list items as you consume them. Iterators can handle all this for you (without the hefty O(n) tax for those list deletions) 

```
    textStream = iter(re.split(r, x))
    for item in textStream:
        plaintext = item
        url = next(textStream, None)
        # no additional bookkeeping required
        write(plaintext)
        if url:
            write(url)
```

Reply:
done. also added mailto: URLs while at it